django_components/component.py

```diff
@@ -60,9 +62,93 @@ def __new__(mcs, name, bases, attrs):
             if hasattr(media, "js") and isinstance(media.js, str):
                 media.js = [media.js]
 
+        _resolve_component_relative_files(attrs)
+
         return super().__new__(mcs, name, bases, attrs)
 
 
+def _resolve_component_relative_files(attrs: dict):
+    """
+    Check if component's HTML, JS and CSS files refer to files in the same directory
+    as the component class. If so, modify the attributes so the class Django's rendering
+    will pick up these files correctly.
+    """
+
+    # Prepare all possible directories we need to check when searching for
+    # component's template and media files
+    candidate_dirs = settings.STATICFILES_DIRS or []
```

Comment:
NOTE: So in my initial code, I relied only on `settings.TEMPLATES`. But I've seen in this codebase that in the `sampleproject/settings`, the components are defined via `STATICFILES_DIRS`. So I've included in `STATICFILES_DIRS` among the looked up directories too.

However, when we run `component.render(...)`, then Django's templating engine cannot find the files unless we specify them in the `TEMPLATES.DIRS` setting too.

So that means that adding `STATICFILES_DIRS`  to `candidate_dirs` doesn't really make sense, because the same dirs must be set also in `TEMPLATES.DIRS` for things to work.

So I plan to remove this line, but wanted to share this.

Reply:
I'm afraid lots of people have touched that code over the years, and that no one (including me) has a clear grasp of how it's supposed to work.

Some things to consider:
* The point of this library is to bundle django templates and static files, and this goes against the standard way to structure Django projects, namely putting templates in TEMPLATES.DIRS and static files in STATICFILES_DIRS. I'm not sure what the recommended way should even be, given that we ideally should keep supporting the collectstatic command, and the component.render doesn't work with STATICFILES_DIRS as you said.
* I was under the impression (could very well be wrong), that adding 'django_components.template_loader.Loader' as a template loaded made it possible to make it possible to load templates where ever we wanted. This seems not to be true?
* In the README, under [installation](https://github.com/EmilStenstrom/django-components/?tab=readme-ov-file#installation), we advise people to set the STATICFILES_DIRS settings, so I'm guessing that this is what people have set. It would be good to take backwards compatibility into account here.

As the person that last read the code, who likely has the best knowledge of how the code works, what would you recommend?

Reply:
For the `candidate_dirs` would it make sense to use the same directories as the ones found during in `autodiscover()`? https://github.com/EmilStenstrom/django-components/blob/7058f05e0415965ddd162d0ba663c5106e7ffb66/django_components/__init__.py#L27

Or is that leaving out some directories?

I think it might be more consistent. It could be something like those dirs + staticfiles_dirs if that's needed, but I guess based on what @JuroOravec is saying it might not be the case.

Reply:
Yes, using the same directories everywhere sounds like a good idea.